Epic/bopis

surya-prakash-singh1 opened the pull request:
As part of the 6.0 release, users will be allowed to select a store to pick items. A new Pickup In Store module is added which can be added in the storefront to support this.

SVendittelli committed:
feat: add pickup in store feature (#15765)

Add a feature-lib for Buy Online, Pickup in Store.

Co-authored-by: Michael, Christopher <chris.michael@sap.com>
Co-authored-by: Singh, Surya <surya.singh@sap.com>
Co-authored-by: Pucek9 <pucek9@gmail.com>

CXSPA-6

surya-prakash-singh1 committed:
Merge branch 'develop' into epic/bopis

surya-prakash-singh1 committed:
feat: add BOPIS modal window (#15794)

Add a feature for pickup in store modal with data exchange and styling.

CXSPA-489

Co-authored-by: Sam Vendittelli <sam.vendittelli@sap.com>
Co-authored-by: Surya Singh <surya.singh@sap.com>

SVendittelli committed:
CXSPA-9 Implement Store List component (#15913)

* refactor: Enable strict mode for storefrontlib (#15770)

* feature/CXSPA-390 (#15796)

fixed my company drop down arrow

* fix: minimum width and height adjustment to buttons and links on mobile (#15784)

* feature/CXSPA-440
Added min height at mobile screen size for elements that were not passing light house seo test

* feature/CXSPA-440
added more changes for link and button min height and width

* feature/CXSPA-440
prettier clean up

Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>

* feat: add child component to productDetailsTab (#15808)

A child component is injected within ProductDetails tab, below the product details text. 
This change has been made to allow PDF component to be displayed below product details.

* style: fix product listing buttons (#15659)

* feature/CXSPA-233
updated button sizes

* feature/CXSPA-233
fixed button sizes

* Trigger Build

* Trigger Build

* Fixed width issue of buttons

* Fixed failing test cases

* Trigger Build

* Trigger Build

Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>
Co-authored-by: Dalvir Singh <dalvir.singh@sap.com>

* fix: Lock CI jobs to use angular 13 instead of the latest tag (#15821)

context:

- the ci workflow contained some jobs that needed angular to be installed such as lighthouse, unit test lib, and unit core. However, they downloaded the angular/cli with the @latest tag. Reason being, we wanted the pipeline to run with the latest angular and not a specific angular 13 minor/patch.
- 5 days ago the latest tag was set to 14.0.0, therefore causing issues to our pipeline.

closes https://jira.tools.sap/browse/CXSPA-565

* test: fix strict mode issues in attribute related tests (#15819)

closes #15818

* feat: New schematics installer mechanism (#15459)



closes #15380

* feat: support attribute types with additional values (#15699)

 4 additional attribute types: Dropdown and radio group with additional alphanumeric or numeric values
Closes #15688

* feat: Add to cart from order history and saved cart (#15736)

Add Buy It Again from order history
Add Add to Current Cart from saved cart

closes https://jira.tools.sap/browse/CXSPA-308

* chore: Re-wire the events in checkout for better readability and tracing (#15789)

closes https://jira.tools.sap/browse/CXSPA-519

* feat: integrate with store finder service (#15820)

feat: integrate with store finder service

CXSPA-544

* chore: migration schematics and documentation for new configurator attribute types

closes #15689

* feat: Merchant can customize Product Carousels with inner components ex. Add To Cart  (#15817)

* Add add-to-cart to product carousel

* Cannot use absolute path outside of spec files - exporting ProductCarouselComponent sub-component

* Using mock ProductCarouselItem for product-carousel tests and moving UI tests to product-carousel-item
Providing type for Product
Removing unnecessary config in product-carousel module as it goes unseen by the customer anyway
SCSS for showing multiple CMS components in product-carousel-item

* E2E test for customizing Product Carousel component with Add to Cart button

* Setting the ChangeDetectionStrategy for ProductCarouselItemComponent to OnPush

* Adding documentation for breaking change for ProductCarouselComponent

Updating migration docs to have selector of new CarouselItemComponent

* Provide empty string for the Carousel Item's image if the product has no name

Co-authored-by: i822892 <arkadiy.sukharenko@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

* fix: Move cart core code out of the 'main' chunk (#15816)

closes https://jira.tools.sap/browse/CXSPA-542

* add password visibility test (#15812)

* add password visibility test

* password visibility test

* changes requested for pwd visibility

* removed duplicated clear from password test

Co-authored-by: Brian Gamboc-Javiniar <brian_javiniar@hotmail.com>

* test: Password visibility e2e changes

* chore: Adjust CPQ user (#15829)

Adjust CPQ user and its credentials for cypress tests

Closes #15828

* doc: Update schematics readme (#15841)

This PR updates the internal schematics README.md file.

* fix: wait with bootstraping the app until OAuth params are checked in the URL (#15831)

### Context
When OAuth implicit or code flow is configured, on page start (especially when returing from the external auth server to Spartacus), we attempt to retrieve the access token from the URL, with the method `authService.checkOAuthParamsInUrl()`. During the execution of this method, the long OAuth authorization_code handshake can happen. Therefore the access token might be retireved with a delay.

### Before fix
Previously we didn’t wait until the access token is retireved, but started bootstrapping the application right away, including running of the route guards. Especially it was possible to run the AuthGuard, before we retireved the access token by the method `authService.checkOAuthParamsInUrl()`. This could result in AuthGuard blocking the navigaiton becasue the user was **not yet** logged in (however awaiting the access token in the background).

### After fix
Now we wait for `authService.checkOAuthParamsInUrl()` to resolve, and only then we proceed with bootstrapping the application and running the guards, including `AuthGuard`.

fixes #13622
fixes CXSPA-405

* feature/CXSPA-524 (#15815)

updated styling for mobile viewport of table header

* test: fix flaky VC e2e tests

Closes #15827

* feature/CXSPA-298 (#15810)

Fixed tablet size input check box alignment

Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>

* chore: install VC configurator by default in the internal installation script (#15844)

Previously the Variant Configurator was not installed by default in our internal installation script `/install/run.sh`. It was installed implicitly only in case when installing CPQ extension (with `ADD_CPQ` flag).

Since the the PR https://github.com/SAP/spartacus/pull/15459 was merged, the CPQ installation is aborted if VC was not installed explicitly before.

In this PR we update the installation script to: install the Variant Configurator first explicitly (by default), and only then conditionally (with `ADD_CPQ` flag) install CPQ extension on top of it.

fixes https://jira.tools.sap/browse/CXSPA-588

* CXSPA-545 BOPIS Store List (#15852)

Add store list components.

CXSPA-545

Co-authored-by: Christopher Michael <chris.michael@sap.com>
Co-authored-by: mechris3 <mechris3@hotmail.com>
Co-authored-by: cawmichae <cawmichae@users.noreply.github.com>
Co-authored-by: I554698 <surya.singh@sap.com>

* refactor: Enable strict mode for core lib  (#15807)

* refactor: Remove media component deprecations from 4.2 (#14236)

Deprecation removals for #13172

* feat: CXSPA-600 Remove deprecated errors$ field (#15867)

* feat: Enhance overview normalizer to display value price (#15866)

Enable displaying value prices on the overview page for VC products

Closes: #15840

* fix: Spinner in guest checkout after event refactoring (#15870)

fixes guest checkout infinite loop in delivery address after checkout event refactoring

https://jira.tools.sap/browse/CXSPA-603

* feature/CXSPA-591 (#15859)

fixed import products modal button

* chore: lint integration libs with i18n-lint (#15877)

`i18n-lint` checks if no hardcoded texts appear in the HTML templates

Previously integration-libs were not linted, but only feature-libs and the storefrontlib. Now integration libs are also linted.

Btw. removed the `--exclude` of node_modules folder, as being redundant - node_modules don't overlap with any allowlisted files.

fixes https://jira.tools.sap/browse/CXSPA-627

* chore: AuthRedirectService methods deprecation cleanup and migration (#14481)

Closes: #14459

* chore: skip flaky B2B saved cart E2E test for now

* chore: improve guest checkout e2e tests (#15876)

* Changed font-weight and size to fix the amp button contrast

Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>

* style: change order replenish table styling to match my company table

* style: Added padding to the order date

Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>

* chore: Remove deprecated notices for vouchers. (#15864)

* chore: Remove deprecated styling class for quick order form component

* chore: Protect deprecated cmfRef in ComponentWrapperDirective (#15878)

* refactor: Enable strict mode for organization lib

* chore: remove deprecated code from OccEndpoints model

closes https://jira.tools.sap/browse/CXSPA-597

* chore: User service deprecation

* chore: Remove deprecated keyboard focus config and on navigate focus … (#15871)

* fix: Site freezes after agent logs out of ASM (#15880)

* Revert "Site freezes when logging out of ASM (#14914) (#15375)"

* Explicitly set logout process back to false after the operation has completed

* Create e2e and unit tests to verify new logout behavior

* Create duplicate e2e util functions for waitForPage and waitForCategory in central navigation.ts. To be consolidated in future ticket.

* Improving billing address e2e tests (#15891)

* Removing it blocks

* Asserting on registration is not in scope for this test

* No need to enter this addres again

* Using a shortcut to register and login test user

* Trigger Build

* Trigger Build

Co-authored-by: KateChuen <45409515+KateChuen@users.noreply.github.com>
Co-authored-by: KateChuen <kate.chuen.cheong@sap.com>

* fix: Allow to place order during midnight time (#15901)

- remove date bug, where 24h time for midnight translates to 24:xx instead of 00:xx. Basically, backend for the replenishment endpoint does not accept 24 for midnight, but 00 / reference to fix conversion https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/hourCycle

* test: Enhance checkout replenishment e2e (#15887)

included fixtures for the checkout replenishment e2e and testing with premium delivery instead of standard delivery like all our other checkout e2e does

closes https://jira.tools.sap/browse/CXSPA-530

* test: Fix anonymous consent config e2e (#15902)

Change the user journey in the e2es by making user anonymous consents gets loaded before creating the user and navigating to the consent page.

closes https://jira.tools.sap/browse/CXSPA-653

* feat: Fetching products in carousel with stock information (#15838)

* Fetching products in carousel with stock information

* As PRODUCT_SCOPE in ProductCarouselComponent has been changed into a list, using other scopes when fetching products is redundant

* Rewriting unit test for ProductCarouselComponent to use done() callback for assertions in subscription

* Trigger Build

* Adding new ProductScope for stock information only - using this scope instead of DETAILS for ProductCarousel

Co-authored-by: pla <plabadie@users.noreply.github.com>

* chore: improve clear cart core e2e tests (#15886)

* CXSPA-554 Integrate store stock level (#15912)

Integrate pickup in store with the stock level API.

Co-authored-by: I554698 <surya.singh@sap.com>
Co-authored-by: mechris3 <mechris3@hotmail.com>
Co-authored-by: cawmichae <cawmichae@users.noreply.github.com>

* fix: Lint issue

* test: Improve currency checkout page core e2e (#15909)

closes https://jira.tools.sap/browse/CXSPA-656

* chore: fix "Public API Changes" bot (#15883)

### Improvements
- make the "Public API changes" bot back working in CI (get rid of all the errors in the script)
  - side effect: reduce the time of executing "Public API changes" bot from 2h back to ~30min (see [test PR](https://github.com/SAP/spartacus/actions/runs/2580268027))
- reduced the warnings noise in logs:
  - disabled API-extractor warnings not relevant in our project, e.g. [warning for missing release tags like alpha/beta/internal/public in our TSDocs](https://api-extractor.com/pages/messages/ae-missing-release-tag/),
  - disabled API-extractor warnings about missing exports, because sometimes caused false-positive warnings e.g. when an interface is exported, but from a different library
  - disabled all warnings from TSDocs parser (checking the syntax of TSDocs is not in scope of our breaking changes detection bot; generated ~30k warnings per script execution before)
- reduced the warnings noise in the report comment:
  - API Extractor warnings are not printed to the report comment, but only in the console of the script - to avoid fake reports because of a different file path, for example:
     ```diff
     -// /github/workspace/branch-clone/dist/<...file path> - <warning message>
     +// /github/workspace/dist/<...file path> - <warning message>
    ```

### Changes
- analyze d.ts instead of TS files in dependent libraries (e.g. when finding `import {X} from '@spartacus/core'`, analyze `dist/core/.../.d.ts` instead of `/projects/core/.../.ts`)
- disable api-extractor warning of type `ae-missing-release-tag` and `ae-forgotten-export`
- disable all TSDoc warnings (of type `tsdocMessageReporting`)
- disable reporting warnings in the markdown report, but leave them in the script console output
- add explicit return type for `APP_INITIALIZER`s that returned `()=>Promise<Config>` - to avoid api-extractor error `ERROR: Failed to fetch entity for import() type node: import("@spartacus/core").Config`
  - by the way, I added also missing return types for other `APP_INITIALIZER`s, which was not essential for this PR
- by the way, removed some redundant JSDoc notices `@member` from `occ-models.ts` - which caused warnings in api-extractor and were redundant

### Implementation notes
Previously for API Extractor tool we used the tsconfig with path mappings to source code (e.g. for resolving an import from `@spartacus/core` it looked up the source code TS files in `projects/core/*`).

According to https://api-extractor.com/pages/messages/ae-wrong-input-file-type/ , it seems that we were using the api-extractor tool not in a way that the tool's authors assumed. We analyzed TS files (only in dependency packages), but api-extractor expects to analyze d.ts

Moreover, since API Extractor v7.24.0 (released 2022-05-14), analyzing TS files throws an error (see [changelog](https://github.com/microsoft/rushstack/blob/main/apps/api-extractor/CHANGELOG.md#7240) and [troubleshooting page](https://api-extractor.com/pages/messages/ae-wrong-input-file-type/)), which caused our CI script for api-extractor to fail since 2022-05-14.

In this PR, we start using the "production" version of `tsconfig` (`tsconfig.app.prod.json` instead of `tsconfig.json`), which uses `"paths"` mapping to the `/dist` folder, which contains d.ts files.

* doc: define the max version of Angular version per Spartacus version (#15001)

* chore: fix flaky cart e2e test (#15916)

* chore: add important note in GH release template to avoid unnecessary commits and tags

* chore: add important notes for new releases

* Adding instructions for core tag

* refactor: make anchors compliant with checkmarx (#15932)

Anchors tags have been refactored to be compliant with the new pipeline security check.
Attribute 'rel' has been added to anchors that are using _blank as target.

* test: Improve asm flaky e2es (#15929)

Notes:

- removed redundant registration that happens twice at the beginning and other duplicate tasks
- make sure to wait for asm input box to no longer be disabled before typing
- removed unused functions
- removed duplicates and reuse the intended function for customer emulation tests
- instead of visiting a page by using cy.visit, we use the my-account dropdown instead. Visiting pages is not a user journey on how a user would navigate the page. Also, visiting a page is like a refresh as it's changing the url and going to that route through the browser instead of making it a user journey (an end to end test).
- in addition, the visit caused most of the customer agent session timeout because of some existing bug in the feature

---

important:

- reproducible asm agent session timeout as shown in videos by just waiting ~6 mins in a page. Needs fix in feature for token expiration instead, however, the videos in the issue shows that flakiness as well (not related to session timeout), which should be fixed by this PR

closes https://jira.tools.sap/browse/CXSPA-661

* chore: Spinner on use my location, pick up in store button etc

* fix: remove cart with invalid products (#14228) (#15548)

* changes to addEntries

* added test case

* delete cart test update

* updated test

* test refactor

* account for UNKNOWN_ERROR as well

* add test for UNKNOWN_ERROR

* chore: Changes for New Schematics and otherminor changes

* fix: yarn config updates

* fix: add .scss suffix in schematic

* fix: Schematics and Test Cases

* fix: Schematics Test for pick up in store

* fix: Schematics Test for pick up in store Snapshot updates

* chore: fix sementic test case

* fix: test case

* feat: remove storefinder dependency

* fix: Test Case

Co-authored-by: Weizheng Gao <44440575+WeizhengSap@users.noreply.github.com>
Co-authored-by: morganm58 <mikel.morgan@sap.com>
Co-authored-by: Miguel Estrada <15113219+developpeurweb@users.noreply.github.com>
Co-authored-by: Florent Letendre <florent.letendre01@sap.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>
Co-authored-by: Dalvir Singh <dalvir.singh@sap.com>
Co-authored-by: Brian Gamboc-Javiniar <brian_javiniar@hotmail.com>
Co-authored-by: Christoph Hinssen <33626130+ChristophHi@users.noreply.github.com>
Co-authored-by: Nikola Zarić <nikola.zaric.ns@gmail.com>
Co-authored-by: Parthlakhani <parth.lakhani@sap.com>
Co-authored-by: justinlee01 <47992455+justinlee01@users.noreply.github.com>
Co-authored-by: i822892 <arkadiy.sukharenko@sap.com>
Co-authored-by: KateChuen <45409515+KateChuen@users.noreply.github.com>
Co-authored-by: Gilberto Alvarado <gilberto.alvarado.jimenez@sap.com>
Co-authored-by: LarisaStar <61147963+Larisa-Staroverova@users.noreply.github.com>
Co-authored-by: Krzysztof Platis <platonn.git@gmail.com>
Co-authored-by: Christopher Michael <chris.michael@sap.com>
Co-authored-by: mechris3 <mechris3@hotmail.com>
Co-authored-by: cawmichae <cawmichae@users.noreply.github.com>
Co-authored-by: I554698 <surya.singh@sap.com>
Co-authored-by: Caine Rotherham <rotherham.c@gmail.com>
Co-authored-by: pawelfras <fras.pawel@yahoo.com>
Co-authored-by: Patryk Smul <psmul@divante.pl>
Co-authored-by: Radhep Sabapathipillai <34665674+RadhepS@users.noreply.github.com>
Co-authored-by: pla <plabadie@users.noreply.github.com>
Co-authored-by: Mateusz Kolasa <mateusz.kolasa@sap.com>
Co-authored-by: Jeremy Laverdiere <88747151+bullfroggy-sap@users.noreply.github.com>
Co-authored-by: KateChuen <kate.chuen.cheong@sap.com>
Co-authored-by: Andi Braimllari <101869667+I539321@users.noreply.github.com>

surya-prakash-singh1 committed:
Merge branch 'develop' into epic/bopis

SVendittelli committed:
CXSPA-22 [BOPIS] Store the consumer's preferred store in the browser (#15971)

* feat: store user's preferred store

* feat: integrate consents

CXSPA-22

Co-authored-by: cawmichae <cawmichae@users.noreply.github.com>

surya-prakash-singh1 committed:
Merge remote-tracking branch 'origin/develop' into epic/bopis

surya-prakash-singh1 committed:
chore: change end to flex-end

SVendittelli committed:
CXSPA-698 [BOPIS] Implement Pickup functionality on PDP (#16019)

* CXSPA-724 [BOPIS] Create slice of store for intended pickup locations (#15977)

* feat: add intended pickup locations

* test: pickup locations redux store

* CXSPA-732 [BOPIS] Create service layer for intended pickup locations (#15984)

* feat: add intended pickup location service

* feat: clear intended location on picking delivery

* feat: close pickup dialog on store selection

* feat: default options to pickup if they have an intended location for that product in the store

* feat: feature toggle for pick up in store Target version:5.2

* feat: add pickupStore to cart entries API

* CXSPA-752 [BOPIS] Check for intended location before adding to cart (#15997)

* feat: Disable Pick Up From Here if Store does not have stock for selected product

* feat: Show pickup delivery options on products marked availableForPickup: true

* test: add to cart pickupStore

* test: pickup delivery options

* test: store component template

* CXSPA-21 [BOPIS] Display Pickup/Delivery options on the PDP (#16018)

* CXSPA-798 Cypress tests (#16012)

Co-authored-by: I554698 <surya.singh@sap.com>
Co-authored-by: surya-prakash-singh1 <surya-prakash-singh1@users.noreply.github.com>
Co-authored-by: cawmichae <cawmichae@users.noreply.github.com>
Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>
Co-authored-by: mechris3 <mechris3@hotmail.com>

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

SVendittelli committed:
CXSPA-886 Replace geolocation on page load with only preferred store (#16031)

* feat: implement stock level at store API

* feat: add ngrx state for stock at store

* feat: create connector/adapter/facade/store layers

* feat: check intended pickup location

* feat: keep preferred store displayName in local storage

* feat: only show store dialog if needed

* test: replace geolocation with preferred store

Co-authored-by: mechris3 <mechris3@hotmail.com>

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

SVendittelli committed:
CXSPA-798 [BOPIS] Test_TC: Check the radio buttons and dialogue for PDP (#16032)

* test: add more complete E2E tests

Co-authored-by: mechris3 <mechris3@hotmail.com>

Commit:
chore: update checklist and todos

Commit:
fix: change colon to pipe

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

SVendittelli committed:
feat: add delivery option outlet to checkout (#16037)

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
CXPSA-921 [BOPIS] Split smart and display component for pickup options (#16053)

* refactor: reorganise bopis components

* feat: add options component to wrappers

* refactor: move logic to container component

* feat: add pickupOption to state for productCodes

* fix: new Cypress, fix failed case, fix unit test

Co-authored-by: mechris3 <mechris3@hotmail.com>
Co-authored-by: I554698 <surya.singh@sap.com>

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
CXSPA-752 [BOPIS] Implement Pickup functionality on PDP (#16075)

* feat: add cart wrapper to pickup in store

* test: improve PickupOptionsComponent tests

Document the PickupOptionsComponent and extend unit tests to cover the
template.

* docs: improve bopis documentation

* refactor: remove unused dependency

* feat: move bopis behind env feature flag

* fix: correct in stock checks

* test: new out-of-stock logic

* test: bopis active cart service

* fix: remove function calls from template

To prevent repeated execution of functions on changes, extract the out
of stock logic to utility and set variable on init.

Additionally, default bopis to be on for development purposes.

* fix: remove circular dependency

* test: correct cypress expectations

As low stock is not available for pickup, we need to change the store we
use in our cypress tests.

CXSPA-752

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
CXSPA-1114 [BOPIS] Add copyright headers (#16079)

* chore: add copyright headers

To ensure that we have no SonarQube issues relating to licences, we need
copyright headers for all our ts files.

These are generated with the `./ci-scripts/prepend-license.sh` script.

* chore: add comment to empty constructors

SonarQube highlights empty constructors as a code smell, so we are adding
comments to them to indicate we intend for them to be empty.

* chore: remove debug statements

CXSPA-1114

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
fix: update mini cart to count pickup items (#16092)

The original implementation of the mini-cart component only counted
products being delivered. Using the total unit count also includes items
being picked up too.

CXSPA-12

JedidiahAwuku committed:
CXSPA-1115 [BOPIS] Create order details presentational component (#16094)

* feat: Add scaffolding for pickup delivery info

* feat: Add basic HTML structure

* feat: Add component for address and schedule.

* test: Add unit test

surya-prakash-singh1 committed:
CXSPA-921 [BOPIS] Split smart and display component for pickup options (#16091)

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
chore: update unit test scripts for junit reporter

Commit:
CXSPA-1233 [BOPIS] Export unit test results to junit xml file (#16123)

Fix test errors for missing selectors

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

surya-prakash-singh1 committed:
CXSPA-11 [BOPIS] Display Delivery/Pickup Options on Cart page (#16112)

* refactor: CartPickupOptionsContainerComponent

* fix: incorrect entry number for duplicate item in cart

* refactor: setPickupOption names and types

* fix: update preferred store from cart options

* feat: Default pickup location to preferred store if it has stock

* fix: remove unused dependencies

* fix: derive cart pickup option from store

Co-authored-by: Sam Vendittelli <sam.vendittelli@sap.com>

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

surya-prakash-singh1 committed:
CXSPA-1225 [BOPIS] Create automated tests for logged-in user checkout for BOPIS (#16135)

* test: Adds Cypress Test Cases

cawmichae committed:
Feature/cxspa 1225 (#16154)

* feature: Add test for bopis logged in user

* feature: Finish Test apart from TODOs

Co-authored-by: mechris3 <mechris3@hotmail.com>

cawmichae committed:
Feature/cxspa 1263 (#16167)

* feat: Add test for guest user checkout journey

cawmichae committed:
Merge branch 'develop' into epic/bopis

cawmichae committed:
task: Remove index.ts files and move imports into public_api.ts files

cawmichae committed:
feat: Add Cypress Test - Change product from picku (#16203)

* feat: Add Cypress Test - Change product from picku

* feat: update deps

Commit:
[BOPIS] refactor (#16137)

* refactor: unify constructor parameter types

For extensibility and conformity with other feature libs, migrate
constructor parameters to be protected in preference to private and
readonly.

* refactor: rename the PickupOptionDialogComponent

* refactor: rename PickupInfoComponents

* refactor: pickup option dialog

* refactor: sort params, imports, exports

Sort constructor parameters, imports, and exports alphabetically.

* chore: improve jsdocs

* refactor: remove unused service

* fix: correct missed rename

* refactor: remove readonly from constructors

* chore: complete angular 14 upgrade

Commit:
test: fix bopis schematics unit tests (#16206)

cawmichae committed:
feat: Add tests for multiple cart entries (#16250)

cawmichae committed:
Merge branch 'develop' into epic/bopis

Commit:
chore: remove obsolete release-it config

cawmichae committed:
Merge branch 'develop' into epic/bopis

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
[BOPIS] feat: give pickup dialog focus on open (#16281)

* feat: give pickup dialog focus on open

When the pickup in store dialog open then give focus to the search box.
Additionally allow the user to close the dialog with the escape key or
by clicking outside it. This is to improve accessability.

* test: update cypress tests for new modal

* test: Fix cypress to incorrectly closing dialog

Co-authored-by: JedidiahAwuku <jedidiah.awuku@sap.com>

JedidiahAwuku committed:
Feature/cxspa 1496 (#16290)

* feat: Add preferred store information to user profile

* chore: Add licence header and update dependencies

JedidiahAwuku committed:
Feature/cxspa 1498 (#16373)

feat: Add set as preferred store feature

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

cawmichae committed:
feature: Fix failing a11y test (#16403)

JedidiahAwuku committed:
Merge branch 'develop' into epic/bopis

JedidiahAwuku committed:
test: Add missing tabbable items to product page (#16416)

Refactor HTML for pickup-options container

Fix cypress test as a result of refactoring HTML for pickup options

JedidiahAwuku committed:
test: Fix missing items from pickup modal tabbing order

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

cawmichae committed:
Merge branch 'develop' into epic/bopis

Commit:
fix: add missing aria attributes to dialog (#16443)

Add the required attributes to indicate that the pickup option dialog is
a modal dialog for screen readers.

Add the role of button to use my location and find all stores as that
is their function, not navigation.

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

surya-prakash-singh1 committed:
fix: Focused items do not have the highlight outline initially when the modal dialog pops up (#16450)

JedidiahAwuku committed:
Feature/cxspa 1499 (#16590)

* feat: Add CMS Component to address Book Content Slot

* feat: Override MyPreferredStore CMS Component

* feat: Remove my-preferred-store folder from storefrontlibs

* Add Preferred Store to Address Book

* chore:Address Book

* feat: Add implementation to method to get direction in my preferred store component.

Move getDirection method from Abstract store item Component to Store Finder service for easier use in my preferred store component.

* Add Unit Tests for My Preferred Store

* Increase Branch coverage of tests

* Test:Cypress for pickup in store  in address book

* Add Make This My Store to Address book

* Add Make This My Store to address book

* Change back env file

* Add New Test, adding chaining operator

* Test:Cypress for pickup in store  in address book

* chore: Run yarn config check
Add license headers

* chore: Apply prettier fix

* feat: Add implementation for getDirectionsToStore in Address book page

* feat(preffered store on store finder page)

Co-authored-by: cawmichae <chris.michael@sap.com>
Co-authored-by: I554698 <surya.singh@sap.com>

cawmichae committed:
Merge branch 'develop' into epic/bopis

cawmichae committed:
Add new PickupInStoreDetails to slot on page

cawmichae committed:
Show DeliverPointOfService and Items

cawmichae committed:
Add merge mao

cawmichae committed:
Get all the data that we need available to us

cawmichae committed:
Move Logic from component to a service

cawmichae committed:
Start Template

surya-prakash-singh1 committed:
feat(pickup in store on delivery mode page)

mechris3 committed:
Test commit

mechris3 committed:
Merge branch 'feature/CXSPA-14' of https://github.com/SAP/spartacus into feature/CXSPA-14

surya-prakash-singh1 committed:
feat(pickup in store on delivery mode page)

surya-prakash-singh1 committed:
feat(pickup in store on delivery mode page)

mechris3 committed:
Refactor DeliveryPointsService

mechris3 committed:
Merge branch 'feature/CXSPA-14' of https://github.com/SAP/spartacus into feature/CXSPA-14

mechris3 committed:
Add Unit tests and some refactoring

mechris3 committed:
Add expects to test

mechris3 committed:
Bug fix in test

JedidiahAwuku committed:
fix: Remove augmented point of service  casting from intended location

JedidiahAwuku committed:
fix: Remove User casting from preferred Store

mechris3 committed:
Add getProperty function to use instead of optional chaining operator.

surya-prakash-singh1 committed:
feat(pickup in store on delivery mode page)

surya-prakash-singh1 committed:
Merge branch 'feature/CXSPA-14' of https://github.com/SAP/spartacus into feature/CXSPA-14

surya-prakash-singh1 committed:
feat(pickup in store on delivery mode page)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

JedidiahAwuku committed:
test: Update store mock in pickup location service spec

surya-prakash-singh1 committed:
Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>
Co-authored-by: Michael, Christopher <chris.michael@sap.com>

surya-prakash-singh1 committed:
Removed: Pickup Info outlet on Cart Page

surya-prakash-singh1 committed:
Fix:(Style for Get Direction button)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout license update)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout lint and config update)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout lint and config update)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

surya-prakash-singh1 committed:
feat(pickup in store on Checkout)

surya-prakash-singh1 committed:
feat(pickup in store on review page)

JedidiahAwuku committed:
feat: Add checkout pick up in store details selector to review pickup in store details component html for reusability

Add store schedule component to checkout pick up in store details selector

Add styles for store schedule component selector to pickup in store details checkout

surya-prakash-singh1 committed:
feat(pickup in store on review page)

JedidiahAwuku committed:
test: Add get directions method to  test  case in StoreFinderModule

Add OutletModule import to test case in StoreFinderModule

surya-prakash-singh1 committed:
feat(pickup in store on review page)

JedidiahAwuku committed:
feat: Add header for product on cart on checkout pickup instore details component

Remove margin for product name

surya-prakash-singh1 committed:
feat(pickup in store on review page)

surya-prakash-singh1 committed:
feat(pickup in store on review page)

surya-prakash-singh1 committed:
feat(yarn config update)

surya-prakash-singh1 committed:
fix(prettier issues)

github-actions[bot] committed:
Add license header

JedidiahAwuku committed:
test: Fix store finder list item component test case

JedidiahAwuku committed:
Merge branch 'feature/CXSPA-17-review' of https://github.com/SAP/spartacus into feature/CXSPA-17-review

surya-prakash-singh1 committed:
feat(yarn config update)

surya-prakash-singh1 committed:
Merge branch 'feature/CXSPA-17-review' of https://github.com/SAP/spartacus into feature/CXSPA-17-review

surya-prakash-singh1 committed:
Merge branch 'feature/CXSPA-14' into feature/CXSPA-17-review

surya-prakash-singh1 committed:
refactor(checkout pickup in store test case)

surya-prakash-singh1 committed:
refactor(checkout pickup in store)

surya-prakash-singh1 committed:
refactor(checkout pickup in store)

surya-prakash-singh1 committed:
refactor(checkout pickup in store)

surya-prakash-singh1 committed:
refactor(checkout pickup in store)

surya-prakash-singh1 committed:
merge(develop to epic)

surya-prakash-singh1 committed:
feat(Update License)

surya-prakash-singh1 committed:
feat(Config Update)

surya-prakash-singh1 committed:
test(fix storefinder duplicate import)

surya-prakash-singh1 committed:
test(angular.json update)

surya-prakash-singh1 committed:
test(angular.json update)

surya-prakash-singh1 committed:
test(schematic update snapshot)

surya-prakash-singh1 committed:
chore(updated todo)

surya-prakash-singh1 committed:
Chore: Split checkout order details review container into presentational and smart components

WeizhengSap commented on ci-scripts/unit-tests.sh:
```diff
@@ -76,6 +76,14 @@ echo "Running schematics unit tests and code coverage for tracking library"
 
+echo "Running unit tests and code coverage for pickup-in-store library"
+
+ng test pickup-in-store --source-map --watch=false --code-coverage --browsers=ChromeHeadless
```
use `--no-watch` and `--browsers ChromeHeadless`
check these:
https://angular.io/cli#boolean-options
https://angular.io/cli#array-options

WeizhengSap commented on ci-scripts/unit-tests.sh:
```diff
@@ -76,6 +76,14 @@ echo "Running schematics unit tests and code coverage for tracking library"
+
+echo "Running schematics unit tests and code coverage for pickup-in-store library"
+
+yarn --cwd feature-libs/pickup-in-store run test:schematics --coverage=true
```
use `--coverage`

WeizhengSap commented on feature-libs/cart/base/components/add-to-cart/add-to-cart.component.html:
```diff
@@ -18,6 +18,9 @@
       >
     </div>
   </div>
+  <ng-container *cxFeatureLevel="'5.1'">
```
confirmed with Bill, the BOPIS will be released in 6.0.

WeizhengSap commented on feature-libs/cart/base/components/add-to-cart/add-to-cart.component.spec.ts:
```diff
@@ -128,6 +129,7 @@ describe('AddToCartComponent', () => {
         SpinnerModule,
         I18nTestingModule,
         ReactiveFormsModule,
+        FeaturesConfigModule,
```
BOPIS will be in 6.0, so no need `FeaturesConfigModule`

WeizhengSap commented on feature-libs/cart/base/components/add-to-cart/add-to-cart.component.html:
```diff
@@ -18,6 +18,9 @@
     </div>
   </div>
+  <ng-container *cxFeatureLevel="'5.1'">
+    <ng-template [cxOutlet]="CartOutlets.ADD_TO_CART_CONTAINER"></ng-template>
```
`ADD_TO_CART_CONTAINER` is confusing. maybe call it `ADD_TO_CART_OPTIONS`?

WeizhengSap replied:
If `ADD_TO_CART_CONTAINER` is used, then maybe make it wrap the counter and stock info:
```
<ng-template [cxOutlet]="CartOutlets.ADD_TO_CART_CONTAINER">
    <div class="quantity" *ngIf="showQuantity">
      <label>{{ 'addToCart.quantity' | cxTranslate }}</label>
      <div class="cx-counter-stock">
        <cx-item-counter
          *ngIf="hasStock"
          [max]="maxQuantity"
          [control]="addToCartForm.get('quantity')"
        ></cx-item-counter>

        <span class="info">
          <span *ngIf="showInventory$ | async">{{ getInventory() }}</span>
          {{
            hasStock
              ? ('addToCart.inStock' | cxTranslate)
              : ('addToCart.outOfStock' | cxTranslate)
          }}</span
        >
      </div>
    </div>
  </ng-template>
```
So, in `PdpPickupOptionsContainerModule`, the `position` can be `after`. Using this way, customer can config the pickup option displayed before or after the counter.

WeizhengSap commented on feature-libs/cart/base/components/add-to-cart/add-to-cart.module.ts:
```diff
@@ -7,12 +7,24 @@
+    I18nModule,
+    ItemCounterModule,
+    OutletModule,
+    FeaturesConfigModule,
```
no need `FeaturesConfigModule`

WeizhengSap commented on feature-libs/cart/base/components/cart-shared/cart-item-list-row/cart-item-list-row.component.html:
```diff
@@ -48,6 +48,14 @@
       <cx-promotions [promotions]="item.promotions"></cx-promotions>
+
+      <!-- Item Delivery Details Outlet -->
+      <ng-container *cxFeatureLevel="'5.1'">
```
![Screenshot 2023-01-11 at 4 09 29 PM](https://user-images.githubusercontent.com/44440575/211907679-6b27675a-80bb-4123-aa49-b433099d44d0.png)
be careful on the UI when there is another cart details addon stuff

WeizhengSap commented on feature-libs/cart/base/core/store/actions/cart-entry.action.ts:
```diff
@@ -41,6 +42,7 @@ export class CartAddEntrySuccess extends StateUtils.EntityProcessesDecrementActi
       cartId: string;
       productCode: string;
       quantity: number;
+      pickupStore?: string;
```
maybe also add this change into unit test?

WeizhengSap commented on feature-libs/cart/base/core/facade/active-cart.service.ts:
```diff
@@ -389,6 +389,7 @@ export class ActiveCartService implements ActiveCartFacade, OnDestroy {
    */
   addEntry(productCode: string, quantity: number): void {
     // TODO(#13645): Support multiple, simultaneous invocation of this function, when cart is not loaded/created
+
```
This TODO comment can be removed (13645 was closed). (I know it is not related to bopis :) )

WeizhengSap commented on feature-libs/pickup-in-store/assets/translations/en/index.ts:
```diff
@@ -0,0 +1,13 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
```
why it has " SPDX-FileCopyrightText: 2022..."?

WeizhengSap commented on projects/storefrontapp/src/app/spartacus/spartacus-features.module.ts:
```diff
@@ -214,6 +219,8 @@ if (environment.s4om) {
     VariantsFeatureModule,
     ImageZoomFeatureModule,
+    ProductConfiguratorTextfieldFeatureModule,
+    ProductConfiguratorRulebasedFeatureModule,
```
they are duplicate lines.

WeizhengSap commented on feature-libs/pickup-in-store/tsconfig.spec.json:
```diff
@@ -0,0 +1,12 @@
+  "compilerOptions": {
+    "outDir": "../../out-tsc/spec",
+    "strict": false,
+    "target": "es2015",
```
`"target": "es2020",`

WeizhengSap commented on feature-libs/pickup-in-store/tsconfig.lib.json:
```diff
@@ -0,0 +1,97 @@
+  "compilerOptions": {
+    "outDir": "../../out-tsc/lib",
+    "declarationMap": true,
+    "target": "es2015",
```
"target": "es2015",

WeizhengSap commented on feature-libs/pickup-in-store/jest.schematics.config.js:
```diff
@@ -0,0 +1,34 @@
+const { pathsToModuleNameMapper } = require('ts-jest');
```
maybe just copy the file from other libs? We made some changes in: https://github.com/SAP/spartacus/pull/16532

cawmichae committed:
Feature/cxspa 2002 order confirmation (#16757)

* feat: Add configure CMS mapping for pickup instore details for confirmation page

* Add BOPIS Info to Order Confirmation

* Remove Delivery Options after order has been placed

* chore(updated license)

* chore(revert env file change)

* Fix failing tests

* chore(removed 2022 SAP License header)

* chore(removed 2022 SAP License header)

* chore(change description for spec)

Co-authored-by: JedidiahAwuku <jedidiah.awuku@sap.com>
Co-authored-by: cawmichael <mechris3@hotmail.com>
Co-authored-by: I554698 <surya.singh@sap.com>

WeizhengSap commented on feature-libs/storefinder/occ/adapters/occ-store-finder.adapter.ts:
```diff
@@ -70,11 +70,11 @@ export class OccStoreFinderAdapter implements StoreFinderAdapter {
   protected callOccFindStores(
     query: string,
-    searchConfig: SearchConfig,
+    searchConfig?: SearchConfig,
```
why `searchConfig` optional?

WeizhengSap commented on feature-libs/pickup-in-store/test-jest.ts:
```diff
@@ -0,0 +1,17 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
we have `setup-jest.ts`, please take a look of other libs.

WeizhengSap commented on projects/core/src/model/product.model.ts:
```diff
@@ -209,3 +211,33 @@ export enum VariantQualifier {
   ROLLUP_PROPERTY = 'rollupProperty',
 }
+
+export interface PointOfServiceStock {
```
can we move these 2 model interfaces into the pickup-in-store lib?

WeizhengSap commented on feature-libs/pickup-in-store/occ/model/occ-stock-endpoints.model.ts:
```diff
@@ -0,0 +1,31 @@
+     * Update the delivery option for a cart entry to pickup or delivery.
+     * @member {string} [page]
+     */
+    updateDeliveryOption?: string | OccEndpoint;
```
this one should be moved to `occ-pickup-location-endpoints.model.ts`, right?

JedidiahAwuku committed:
fix: Mock Store to resolve loadStoreDetails test case (#16766)

mechris3 committed:
Merge branch 'develop' into epic/bopis

cawmichae committed:
Test/cxspa 817 (#16804)

* Finish Cypress Test for A guest user logs in while checking out with BOPIS

JedidiahAwuku committed:
test: Complete cypress test for a logged in user checkout with BOPIS (#16805)

JedidiahAwuku committed:
test: Complete test for cart multiple entries checkout phase (#16806)

Commit:
Merge branch 'develop' into epic/bopis

Commit:
fix(bopis): format files

surya-prakash-singh1 committed:
chore(renaming the flextype to follow convention)

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
chore(bopis): update ts config

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

surya-prakash-singh1 committed:
chore(refractor Pickup In Store) (#16925)

Commit:
[BOPIS] CXSPA-906 Code splitting (#16932)

* Split Code into Eager and Lazy Load Module

* Update dependencies in preperation for code splitting

* Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>
Co-authored-by: Surya Prakash <surya-prakash-singh1@users.noreply.github.com>
Co-authored-by: Vendittelli, Sam <sam.vendittelli@sap.com>

* Largely Working

Co-authored-by: Eric Polesky <ercultimate@users.noreply.github.com>
Co-authored-by: Surya Prakash <surya-prakash-singh1@users.noreply.github.com>
Co-authored-by: Vendittelli, Sam <sam.vendittelli@sap.com>
Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>

* fix: Resolve issue with yarn build:pickup-in-store

Co-authored-by: Michael, Christopher <chris.michael@sap.com>

* Not working, but split well ish

* Commit so Sam can build on a MAC!

* feat(pickup-up-in-store): fix build issues from code splitting

* test(pickup-in-store): fix broken tests

Some tests were using the incorrect paths after code splitting. Correct
the paths and add CMS config to the module unit test.

* feat(pickup-in-store): flatten base

As we do not have sub-features of pickup-in-store, having the extra base
layer is redundant.

* test(pickup-in-store): fix schematics tests

* fix(pickup-in-store): fix module name for schematics generation

* chore(pickup-in-store): rename pickupOptionService usages

* chore(pickup-in-store): remove base references

* feat(pickup-in-store): add CMS component names

---------

Co-authored-by: cawmichael <mechris3@hotmail.com>
Co-authored-by: Eric Polesky <ercultimate@users.noreply.github.com>
Co-authored-by: Surya Prakash <surya-prakash-singh1@users.noreply.github.com>
Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>
Co-authored-by: JedidiahAwuku <jedidiah.awuku@sap.com>
Co-authored-by: Michael, Christopher <chris.michael@sap.com>

Commit:
Merge remote-tracking branch 'origin/develop' into epic/bopis

Commit:
fix(pickup-in-store): update bopis scripts to use npm

mechris3 committed:
Merge branch 'develop' into epic/bopis

surya-prakash-singh1 committed:
Co-authored-by: JedidiahAwuku <JedidiahAwuku@users.noreply.github.com>
Co-authored-by: Michael, Christopher <chris.michael@sap.com>

surya-prakash-singh1 committed:
chore(config update)

surya-prakash-singh1 committed:
chore(config update)

surya-prakash-singh1 committed:
fix(lint)

surya-prakash-singh1 committed:
fix(schematic test)

surya-prakash-singh1 committed:
Merge branch 'develop' into epic/bopis

JedidiahAwuku committed:
Feature/cxspa 2659 order history wip (#17042)

* refactor: Disable shipping address step if there is no delivery entries in cart (#16642)

* refactor: Display pickup in store items or delivery method conditionly (#16739)

* refactor: make entry functions comply with cart entry OCC API (#16769)

* refactor: Only display delivery method if there is no pickup items in cart (#16789)

* feat: add review shipping CMS component (#16808)

* add checkout review shipping component

* remove css

* add active cart

* Add license header

* use new active cart function

* styling

* displaying address, mode, and cart items

* matching wireframes

* clean up components

* fix styling for shipping cpmt

* prettify

* add unit tests

* changes

* add styles

* keep css to not break b2b checkout

* remove img class

* rearrange unit test


* change to shipping label

* change to cms component

* revert checkout submit changes


* update component

* add back review-submit css file

* feature/cxspa-1800: add some test slot


* use deliveryEntries text

* remove placeOrder

* remove step usage to determine display

* update to use totalUnitCount and remove header

* add on push strategy

* add to review overview

* feat: add review payment CMS component  (#16813)

* refactor: output component/template reference in outlet directive (#16830)

* refactor: refactor order details and order confirmation page (#16810)

* refactor: pass pickup store name from outlet component to AddToCartComponent (#16849)

* refactor: add sample data (#16841)

* add getPickupEntries function in facadefactory

* refactor: remove some duplicate codes (#16900)

* feat(order history with bopis)

* fix(buy it again)

* style: Add icon for buy it again action in order history page

Change text color for buy it again  to match wires for order history
page

* test: Add test for normalizing format of address
in order consigned entries component.

* chore: Add required strings for i18N in order consigned entries component

* chore: fix checkout refactor e2e tests (#16993)

* feat: Disable radio buttons to prevent ambiguous entry error
when pickup option changes on cart page

Add new test cases and fix broken ones

* chore: run commands on pre push checklist

JedidiahAwuku committed:
chore: Update CX_BASE_URL for dev environment

surya-prakash-singh1 committed:
fix(schematic test)

surya-prakash-singh1 committed:
Merge branch 'epic/bopis' of https://github.com/SAP/spartacus into epic/bopis

surya-prakash-singh1 committed:
fix(sonar issues)

surya-prakash-singh1 committed:
fix(change default value in env file for pickup in store)

surya-prakash-singh1 committed:
fix(change default value in env file for pickup in store)

JedidiahAwuku committed:
fix: change productCode variable in cart pickup
options container to resolve sonarqube issue

mechris3 committed:
Replace yarn with npm

mechris3 committed:
Merge branch 'epic/bopis' of https://github.com/SAP/spartacus into epic/bopis

JedidiahAwuku committed:
fix: add correct commands for scripts in package.json file for pickup instore

surya-prakash-singh1 committed:
Merge branch 'develop' into epic/bopis

surya-prakash-singh1 committed:
Merge branch 'develop' into epic/bopis

JedidiahAwuku committed:
Merge branch 'develop' into epic/bopis

JedidiahAwuku committed:
fix: checkout delivery mode page

JedidiahAwuku committed:
Merge branch 'develop' into epic/bopis
